packages/drivers/odsp-driver/src/createFile.ts

```diff
@@ -255,160 +248,22 @@ export async function createNewFluidFileFromSummary(
-    }
-    const documentAttributes = getDocAttributesFromProtocolSummary(protocolSummary);
-    const attributesSummaryBlob: ISummaryBlob = {
-        type: SummaryType.Blob,
-        content: JSON.stringify(documentAttributes),
-    };
-    protocolSummary.tree.attributes = attributesSummaryBlob;
-    const convertedCreateNewSummary: ISummaryTree = {
-        type: SummaryType.Tree,
-        tree: {
-            ".protocol": protocolSummary,
-            ".app": appSummary,
-        },
-    };
-    const snapshotTree = convertSummaryToSnapshotTreeForCreateNew(convertedCreateNewSummary);
-    const snapshot: IOdspSummaryPayload = {
-        entries: snapshotTree.entries ?? [],
-        message: "app",
-        sequenceNumber: documentAttributes.sequenceNumber,
-        type: "container",
-    };
-    return snapshot;
-}
-
-/**
- * Converts a summary tree to ODSP tree
- */
-export function convertSummaryToSnapshotTreeForCreateNew(summary: ISummaryTree): IOdspSummaryTree {
-    const snapshotTree: IOdspSummaryTree = {
-        type: "tree",
-        entries: [],
-    };
-
-    const keys = Object.keys(summary.tree);
-    for (const key of keys) {
-        const summaryObject = summary.tree[key];
-
-        let value: OdspSummaryTreeValue;
-        // Tracks if an entry is unreferenced. Currently, only tree entries can be marked as unreferenced. If the
-        // property is not present, the tree entry is considered referenced. If the property is present and is true,
-        // the tree entry is considered unreferenced.
-        let unreferenced: true | undefined;
-
-        switch (summaryObject.type) {
-            case SummaryType.Tree: {
-                value = convertSummaryToSnapshotTreeForCreateNew(summaryObject);
-                unreferenced = summaryObject.unreferenced;
-                break;
-            }
-            case SummaryType.Blob: {
-                const content = typeof summaryObject.content === "string" ?
-                    summaryObject.content : Uint8ArrayToString(summaryObject.content, "base64");
-                const encoding = typeof summaryObject.content === "string" ? "utf-8" : "base64";
-
-                value = {
-                    type: "blob",
-                    content,
-                    encoding,
-                };
-                break;
-            }
-            case SummaryType.Handle: {
-                throw new Error("No handle should be present for first summary!!");
-            }
-            default: {
-                throw new Error(`Unknown tree type ${summaryObject.type}`);
+    return createNewFluidContainerCore<ICreateFileResponse>(
+        containerSnapshot,
+        getStorageToken,
+        logger,
+        initialUrl,
+        forceAccessTokenViaAuthorizationHeader,
+        epochTracker,
+        "CreateNewFile",
+        "createFile",
+        content => {
+            if (!content || !content.itemId) {
+                throw new NonRetryableError(
```

Comment:
While the other path does not need itemId (it already has it), does response have same shape? If so, I'd rather remove this callback and inline this check into createNewFluidContainerCore.